### PR TITLE
feat: nested none/all/roles for allowed and blocked secondary roles

### PR DESCRIPTION
**Pull request by sfc-gh-kwasilewski**
Follow-up to 0f0ca926. Removes the TODO about session_policy resource limits: an empty allowed_secondary_roles set could not express “disallow all” secondary roles.

Model each side as a single nested block with exactly one of none, all, or roles set; update docs and examples accordingly.

### CODE REVIEW

File: pkg/resources/session_policy.go [hunk 1]
```diff
@@ -54,35 +54,22 @@ var sessionPolicySchema = map[string]*schema.Schema{
...
-		Elem: &schema.Schema{
-			Type:             schema.TypeString,
-			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
-			DiffSuppressFunc: suppressIdentifierQuoting,
+		Type:        schema.TypeList,
+		Optional:    true,
+		MaxItems:    1,
+		Description: "Specifies the allowed secondary roles for a session policy, if any.",
+		Elem: &schema.Resource{
+			Schema: sessionPolicySecondaryRolesSchema("allowed_secondary_roles"),
```
Comment (sfc-gh-kwasilewski) [inline pkg/resources/session_policy.go:62]: NormalizeAndCompareIdentifiersInSet only works for top-level TypeSet fields, it would no longer work for the nested `roles` field, so I removed it.

File: pkg/testacc/resource_session_policy_acceptance_test.go [hunk 2]
```diff
@@ -267,12 +264,16 @@ func TestAcc_SessionPolicy_BasicUseCase(t *testing.T) {
...
+							planchecks.ExpectDrift(ref, "allowed_secondary_roles.0.none", sdk.String("false"), sdk.String("true")),
+							planchecks.ExpectDrift(ref, "blocked_secondary_roles.0.roles", blockedSecondaryRoles, sdk.String("[]")),
+							planchecks.ExpectDrift(ref, "blocked_secondary_roles.0.all", sdk.String("false"), sdk.String("true")),
 							planchecks.ExpectDrift(ref, "comment", sdk.String(comment), sdk.String(externalComment)),
 							planchecks.ExpectChange(ref, "session_idle_timeout_mins", tfjson.ActionUpdate, externalSessionIdle, sessionIdle),
 							planchecks.ExpectChange(ref, "session_ui_idle_timeout_mins", tfjson.ActionUpdate, externalSessionUiIdle, sessionUiIdle),
-							planchecks.ExpectChange(ref, "blocked_secondary_roles", tfjson.ActionUpdate, sdk.String("[ALL]"), blockedSecondaryRoles),
+							planchecks.ExpectChange(ref, "allowed_secondary_roles.0.none", tfjson.ActionUpdate, sdk.String("true"), nil),
+							planchecks.ExpectChange(ref, "blocked_secondary_roles.0.roles", tfjson.ActionUpdate, sdk.String("[]"), blockedSecondaryRoles),
+							planchecks.ExpectChange(ref, "blocked_secondary_roles.0.all", tfjson.ActionUpdate, sdk.String("true"), nil),
```
Comment (sfc-gh-asawicki) [inline pkg/testacc/resource_session_policy_acceptance_test.go:324]: let's add a test verifying more transitions between the 3 options (all, none, roles).

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: pkg/testacc/resource_session_policy_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-kwasilewski) [reply pkg/testacc/resource_session_policy_acceptance_test.go:324]: Added one additional test step. Here's an up-to-date list of all potential transitions and which test step covers each of them:

1. Roles -> None -- external drift, step 9 
2. Roles -> All -- external drift, step 9
3. Roles -> Empty
4. All -> None -- OK, step 4 
5. All -> Roles -- OK, step 9
6. All -> Empty -- OK, step 5
7. None -> All -- OK, step 4
8. None -> Roles -- OK, step 9
9. None -> Empty -- OK, step 5
10. Empty -> Roles
11. Empty -> All -- OK, step 3
12. Empty -> None -- OK, step 3

I want to keep these tests relatively short, so not all transitions are covered, specifically 3) and 10). Also, let me know if you would like me to add more tests for external changes detection

**Review by sfc-gh-asawicki — APPROVED** [review]

File: pkg/testacc/resource_session_policy_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-asawicki) [reply pkg/testacc/resource_session_policy_acceptance_test.go:324]: Let's ad a few more external drift detection checks (in a follow-up PR). The rationale would be that this is a complex field, and better to be safe than sorry later. 

Question:
`Roles -> Empty` should it be allowed at all? For empty, `None` should be used, right?

Comment (sfc-gh-kwasilewski) [reply pkg/testacc/resource_session_policy_acceptance_test.go:324]: By `Empty` I mean that the field isn't present in the config. The effect differs across fields; e.g., for allowed_secondary_roles, it means all roles are allowed, while for blocked_secondary_roles, it means none are allowed.

Comment (sfc-gh-kwasilewski) [reply pkg/testacc/resource_session_policy_acceptance_test.go:324]: I'll add more external drift detection checks in a follow-up
